pyonfx/utils.py

```diff
@@ -45,10 +45,11 @@ def all_non_empty(
             A list containing lines_chars_syls_or_words without objects with duration equals to zero or blank text (no text or only spaces).
         """
         out = []
-        for obj in lines_chars_syls_or_words:
+        for obj in lines_words_syls_or_chars:
             if obj.text.strip() and obj.duration > 0:
                 out.append(obj)
         return out
+        # return [obj for obj in lines_words_syls_or_chars if obj.text.strip() and obj.duration > 0]
```

Comment:
You can use list comprehension, cleaner.

Reply:
Related to https://github.com/CoffeeStraw/PyonFX/pull/33#discussion_r681132989
The use of list comprehension shows the wrong type I wrote. 

```
Expression of type "list[Line | Word | Syllable | Char]" cannot be assigned to return type "List[Line] | List[Word] | List[Syllable] | List[Char]"
  Type "list[Line | Word | Syllable | Char]" cannot be assigned to type "List[Line] | List[Word] | List[Syllable] | List[Char]"
    TypeVar "_T@list" is invariant
      Type "Line | Word | Syllable | Char" cannot be assigned to type "Line"
        "Word" is incompatible with "Line"
        "Syllable" is incompatible with "Line"
        "Char" is incompatible with "Line"
    TypeVar "_T@list" is invariant
      Type "Line | Word | Syllable | Char" cannot be assigned to type "Word"
  ...PylancereportGeneralTypeIssues
  ```

Reply:
Oh, good to know (I'm learning a lot through your PR).
So, if not already, please remove your comment.